plugins/modules/azure_rm_aks.py

```diff
@@ -113,12 +113,11 @@
                     - 3
     service_principal:
         description:
-            - The service principal suboptions.
+            - The service principal suboptions. If not provided - use system-assigned managed identity.
         suboptions:
             client_id:
                 description:
                     - The ID for the Service Principal.
-                required: true
```

Comment:
The required is not match with argument_spec. The argument_spec is required!

Reply:
I was unsure, if I should remove `required: true` for the suboptions too. Current state of PR seems to work as expected. I understand it so: `linux_profile` is optional, but if `linux_profile:` is given, then suboptions `admin_username` and `ssh_key` must be provided. This is why I kept `required: true` for `admin_username` and `ssh_key`. Or do you mean something else with "The argument_spec is required"?